source/containers.tex

```diff
@@ -6395,18 +6395,17 @@
 
 \indextext{\idxcode{array}!zero sized}%
 \pnum
-\tcode{array} shall provide support for the special case \tcode{N == 0}.
+In the case that \tcode{N} equals \tcode{0},
+\tcode{begin()} equals \tcode{end()} and
+the return value of \tcode{data()} is unspecified.
 
 \pnum
-In the case that \tcode{N == 0}, \tcode{begin() == end() ==} unique value.
-The return value of \tcode{data()} is unspecified.
+In the case that \tcode{N} equals \tcode{0},
+the effect of calling \tcode{front()} or \tcode{back()} undefined.
```

Comment:
Would this be better addressed by putting a precondition on the `front` and `back` functions that `N > 0`.  Does such a precondition fall out of the sequence container requirements already?  If so, this should be a note, not a normative statement.

Reply:
https://eel.is/c++draft/sequence.reqmts#itemdecl:20 `front()` and `back()` dereference iterators.

To me, it seems like a better overall wording strategy is to make explicitly disallow this for empty arrays, like in LWG 2157.

Reply:
Is it intended not to mention `operator[]`?

Reply:
My main point is that if this is already a precondition, the text here should be a Note, not (redundantly) normative.  Do not make `array<T, 0>` seem any more special than it needs to be.